R/widerRepeated.R

```diff
@@ -0,0 +1,50 @@
+#' @name widerRepeated
+#' @title Transform Data Into Wide Format
+#' 
+#' @description Converts a dataframe into wide format given a single REDCap form.
+#' This function assumes that the Records argument is the result of exportRecordsTyped,
+#' and that all empty values have been previously dropped.
+#'   
+#' @param Records A \code{data.frame} containing the records from
+#'        \code{\link{exportRecordsTyped}}
+#' @param rcon A REDCap connection object as created by \code{unlockREDCap}.
+#' @export
+widerRepeated <- function(Records, rcon)
+{
+  idvar <- getProjectIdFields(rcon)
+  ###########################################################################
+  # Check parameters passed to function
+  coll <- checkmate::makeAssertCollection()
+  
+  checkmate::assert_data_frame(x = Records)
+  checkmate::assert_class(x = rcon,
+                          classes = "redcapApiConnection",
+                          add = coll)
+  checkmate::assert_character(x = idvar,
```

Comment:
Remove this call. Since you're using an internal function to get idvar, we have full control over its value. It's also possible for idvar to have length 2, which is appropriate if there's a secondary unique identifier.